### ISSUE TITLE
Classic 6G: haxdfu seems to always fail

### ISSUE THREAD

**Issue by zzanehip** [freemyipod/wInd3x#4]
Hi. Been trying to figure out why I keep encountering "Error: Failed to run wInd3x exploit: string descriptor got unexpected result, wanted 'haxed dfu', got 'USB DFU Device.'" Whenever I try to run the exploit. Device is ipod classic 6th gen. Not sure if this is because of the system I'm running on, or an issue with the device. Thanks.

**Comment by q3k** [comment]
Yeah, this sounds like the bug isn't triggering - but also is probably not crashing the device?

What system are you running this on?

**Comment by InvoxiPlayGames** [comment]
I was having this issue under macOS - the device has to be unplugged and reconnected for the new device string to stick.

**Comment by q3k** [comment]
Sorry, macOS won't be supported until #23 lands.

**Comment by InvoxiPlayGames** [comment]
From what I remember running into it, this issue isn't related to the exploit not working under macOS (it runs perfectly fine - I used the tool to mess with my 6G a fair amount on macOS), libusb on macOS just doesn't update the device string descriptor after the exploit runs. Presumably that issue would persist even with S5Late.
